src/Stan/Hie.hs

```diff
@@ -0,0 +1,36 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Functions to work with @hie@ specific parts.
+-}
+
+module Stan.Hie
+    ( readHieFiles
+    ) where
+
+import HieBin (HieFileResult (hie_file_result), readHieFile)
+import HieTypes (HieFile)
+import NameCache (NameCache, initNameCache)
+import System.Directory (doesFileExist)
+import System.Directory.Recursive (getDirRecursive)
+import UniqSupply (mkSplitUniqSupply)
+
+
+{- | Returns contents of all @.hie@ files recursively in the given
+@hie@ directory.
+-}
+readHieFiles :: FilePath -> IO [HieFile]
+readHieFiles hieDir = do
+    nameCache <- createNameCache
+    hieContent <- getDirRecursive hieDir
+    hieFiles <- filterM doesFileExist hieContent
```

Comment:
Hmm... I think if we also filter by the `.hie` extension, this code should easily work with the `dist-newstyle` and `.stack-work` directories. What do you think?